### PR TITLE
abuseipdb: Feature - Change lookup ip action to support IPv6

### PR DESCRIPTION
**Pull request by import-au**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The manual_readme_content.md has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe):

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers),
  please document them in the `manual_readme_content.md`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `manual_readme_content.md` all methods (eg, OAuth) used to authenticate
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `manual_readme_content.md`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- The API already supports both v4 and v6, just adding v6 to the contains statement so it will accept both formats.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.
- lookup IP only accepts a v4 address today.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.
- lookup ip now accepts ipv4 and ipv6

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by mmomin-crest** [comment]
Hello @import-au ,
I have updated the 'post ip' action to allow both IPv4 and IPv6 addresses, removing the specificity for IPv4. Please let me know if you have any concerns or doubts regarding this change.

**Review by mmomin-crest — APPROVED** [review]

**Review by gdelavadiya-crest — APPROVED** [review]

**Review by mposluszny-splunk — APPROVED** [review]

File: abuseipdb_connector.py [hunk 1]
```diff
@@ -36,6 +37,22 @@ def __init__(self):
...
+        """
+
+        ip_address_input = input_ip_address
+
+        try:
+            ipaddress.ip_address(ip_address_input)
+        except Exception:
+            return False
+
+        return True
```
Comment (bb-splunk) [inline abuseipdb_connector.py]:
```suggestion
        try:
            ipaddress.ip_address(input_ip_address)
        except Exception:
            return False
        return True
```

File: abuseipdb_connector.py [hunk 2]
```diff
@@ -36,6 +37,22 @@ def __init__(self):
         # Call the BaseConnectors init first
         super(AbuseipdbConnector, self).__init__()
 
+    def _is_ip(self, input_ip_address):
```
Comment (bb-splunk) [inline abuseipdb_connector.py]:
```suggestion
    def _is_ip(self, input_ip_address: str) -> bool:
```

File: abuseipdb_connector.py [hunk 3]
```diff
@@ -36,6 +37,22 @@ def __init__(self):
...
+        """Function that checks given address and return True if address is valid IPv4 or IPV6 address.
+
+        :param input_ip_address: IP address
```
Comment (bb-splunk) [inline abuseipdb_connector.py:43]: Adding typing above makes this redundant imho

File: abuseipdb_connector.py [hunk 4]
```diff
@@ -36,6 +37,22 @@ def __init__(self):
         # Call the BaseConnectors init first
         super(AbuseipdbConnector, self).__init__()
 
+    def _is_ip(self, input_ip_address):
+        """Function that checks given address and return True if address is valid IPv4 or IPV6 address.
```
Comment (bb-splunk) [inline abuseipdb_connector.py]: I only see this used below in the context of `ipv6`, so which one is it supposed to validate?

**Review by bb-splunk — CHANGES_REQUESTED** [review]

File: abuseipdb_connector.py (hunk 4 above)
Comment (stutib-crest) [reply abuseipdb_connector.py]: for ipv4 we already have a validation on platform side, but that doesn't validate ipv6 thats why we have added thia function to also validate ipv6 addresses

File: abuseipdb_connector.py (hunk 2 above)
Comment (stutib-crest) [reply abuseipdb_connector.py]: done

File: abuseipdb_connector.py (hunk 1 above)
Comment (stutib-crest) [reply abuseipdb_connector.py]: done

File: abuseipdb_connector.py (hunk 3 above)
Comment (stutib-crest) [reply abuseipdb_connector.py:43]: i didn't exactly understand, you mean to say that the docstring content seems redundant?

File: abuseipdb_connector.py (hunk 4 above)
Comment (bb-splunk) [reply abuseipdb_connector.py]: So why does it say `IPv4 or IPV6` if it only checks for the latter?

File: abuseipdb_connector.py (hunk 3 above)
Comment (bb-splunk) [reply abuseipdb_connector.py:43]: Yeah, with added typing the params description is redundant

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -1,11 +1,11 @@
 repos:
 -   repo: https://github.com/phantomcyber/dev-cicd-tools
-    rev: v1.16
+    rev: v1.20
```
Comment (bb-splunk) [inline .pre-commit-config.yaml]:
```suggestion
    rev: v1.22
```
Is now the latest

**Review by bb-splunk — APPROVED** [review]

File: abuseipdb_connector.py (hunk 4 above)
Comment (stutib-crest) [reply abuseipdb_connector.py]: done

File: .pre-commit-config.yaml (hunk 5 above)
Comment (stutib-crest) [reply .pre-commit-config.yaml]: replaced it with v1.23

File: abuseipdb_connector.py (hunk 3 above)
Comment (stutib-crest) [reply abuseipdb_connector.py:43]: We added that as a part of doc string format.

**Pull request merged**